include/deal.II/fe/fe_values_extractors.h

```diff
@@ -30,7 +30,47 @@ DEAL_II_NAMESPACE_OPEN
  * with corresponding type from the namespace FEValuesViews. There are
  * extractors for single scalar components, vector components consisting of
  * <code>dim</code> elements, and second order symmetric tensors consisting of
- * <code>(dim*dim + dim)/2</code> components
+ * <code>(dim*dim + dim)/2</code> components, as well as second order
+ * nonsymmetric tensors.
+ *
+ * One can think of extractors as the equivalent of an index, or an index range.
+ * In the case of scalar extractors (i.e., the FEValuesExtractors::Scalar
+ * class), creating an object like (see step-20 for this use)
+ * @code
+ *   const FEValuesExtractors::Scalar pressure(dim);
+ * @endcode
+ * can be thought of as creating a single index with value `dim`. By itself,
+ * an index does not know what it is an index to, so it takes the equivalent
+ * of an array to extract anything. Consequently, assume that there is a
+ * finite element with `dim+1` vector components (as indeed there is in
```

Comment:
is not it at least `dim+1`? It could be `dim` velocity, then 1 pressure and `dim` something else, right?